velox/functions/lib/string/StringImpl.h

```diff
@@ -162,6 +162,105 @@ FOLLY_ALWAYS_INLINE int32_t charToCodePoint(const T& inputString) {
   return codePoint;
 }
 
+template <bool isAscii, typename TInString>
+FOLLY_ALWAYS_INLINE int64_t
+hammingDistance(const TInString& left, const TInString& right) {
+  int64_t distance = 0;
+  int64_t leftPosition = 0;
+  int64_t rightPosition = 0;
+  std::string_view leftSV = std::string_view(left.data(), left.size());
+  std::string_view rightSV = std::string_view(right.data(), right.size());
+
+  while (leftPosition < leftSV.size() && rightPosition < rightSV.size()) {
+    int leftSize, rightSize;
+    auto codePointLeft = isAscii
+        ? leftSV.at(leftPosition)
+        : utf8proc_codepoint(&leftSV.at(leftPosition), leftSize);
+    auto codePointRight = isAscii
+        ? rightSV.at(rightPosition)
+        : utf8proc_codepoint(&rightSV.at(rightPosition), rightSize);
+
+    // if both code points are invalid, the behavior is undefined
+    // the following code skips one byte and continues
+    leftPosition += codePointLeft > 0 ? leftSize : -codePointLeft;
+    rightPosition += codePointRight > 0 ? rightSize : -codePointRight;
+
+    if (codePointLeft != codePointRight) {
+      distance++;
+    }
+  }
+
+  VELOX_USER_CHECK(
```

Comment:
use VELOX_USER_CHECK_EQ for better error messages

Reply:
Changed in other places. This one has a conjunction in it

Reply:
> This one has a conjunction in it

Indeed. It would be even cleaner to split this check into two. This way the error message will point out exactly which of the conditions is not met.